File: Selenium/Project/Activity6.py
#Menu checking
#Goal: Make sure that the “Activities” menu item exists and is clickable


# Import webdriver from selenium
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from selenium.common.exceptions import TimeoutException

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service = service) as driver:
  
    # Open the browser to the URL
    driver.get("https://alchemy.hguy.co/crm")

    # Read the title of the website and verify the text
    print("Home page title:"+driver.title)
  
    # Find the username field and enter the username
    driver.find_element(By.XPATH,"//input[@id='user_name']").send_keys("admin")
    # Find the password field and enter the password
    driver.find_element(By.XPATH,"//input[@id='username_password']").send_keys("pa$$w0rd")
    # Find the login button and click it
    driver.find_element(By.XPATH,"//input[@name='Login']").click()
  
    wait = WebDriverWait(driver, 10)  # Wait for a maximum of 10 seconds

    # Locate the "Activities" navigation menu
    activity_menu_is_present = By.XPATH, "//a[@id='grouptab_3']"

    # Ensure that the "Activities" menu item exists
    try:
        wait.until(EC.presence_of_element_located(activity_menu_is_present))
        print("“Activities” menu item exists")
    except TimeoutException:
        print("“Activities” menu item does not exist")

    # Close the browser
    driver.close()

File: Selenium/Project/Activity6.java
/*Menu checking
        Goal: Make sure that the “Activities” menu item exists and is clickable*/

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.TimeoutException;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

import java.time.Duration;

public class Activity6 {
    // Declare the WebDriver object
    private WebDriver driver;
    WebDriverWait wait;
    @BeforeClass
    public void setupMethod() {
        // Set up the Firefox driver
        WebDriverManager.firefoxdriver().setup();
        //Create a new instance of the Firefox driver
        driver = new FirefoxDriver();
        //Open browser and navigate to url
        driver.get("https://alchemy.hguy.co/crm");
    }

    @Test()
    public void menuChecking() {
        //Login the application

        // Find the username field and enter the username
        driver.findElement(By.xpath("//input[@id='user_name']")).sendKeys("admin");

        // Find the password field and enter the password
        driver.findElement(By.xpath("//input[@id='username_password']")).sendKeys("pa$$w0rd");

        // Find the login button and click it
        driver.findElement(By.xpath("//input[@name='Login']")).click();
        String expectedTitle="SuiteCRM";
        System.out.println("Title of the WebPage is:"+driver.getTitle());
        if(driver.getTitle() != null && driver.getTitle().contains(expectedTitle)){
            System.out.println("Home page is opened");
        }
        else{
            System.out.println("Home page could not open.");
        }

        wait = new WebDriverWait(driver, Duration.ofSeconds(10));
        // Locate the "Activities" navigation menu
        By activityMenuIsPresent = By.xpath("//a[@id='grouptab_3']");

        //Ensure that the “Activities” menu item exists
        try {
            wait.until(ExpectedConditions.presenceOfElementLocated(activityMenuIsPresent));
            System.out.println("“Activities” menu item exists");
        } catch (TimeoutException e) {
            System.out.println("“Activities” menu item does not exist");
        }
    }
    @AfterClass
    public void afterMethod() {
        //Close the browser
           driver.close();
    }
}
